src/NLog/Config/ConfigurationItemFactory.cs

```diff
@@ -192,8 +194,8 @@ internal LayoutRendererFactory GetLayoutRenderers()
         [Obsolete("Instead use LogFactory.ServiceRepository.ResolveInstance(typeof(IJsonConverter)). Marked obsolete on NLog 5.0")]
         public IJsonConverter JsonConverter
         {
-            get => _serviceResolver.ResolveJsonConverter();
-            set => (_serviceResolver as IServiceRepository)?.RegisterJsonConverter(value);
+            get => _serviceResolver.ResolveService<IJsonConverter>();
```

Comment:
Missing fallback to static instance?

Reply:
It's added already to the container?

Reply:
So intentional. No need for a default here